File: wallendar-app/app/src/main/java/com/td/wallendar/repositories/mappers/ChargeMapper.java
package com.td.wallendar.repositories.mappers;

import com.td.wallendar.dtos.response.ChargeResponse;
import com.td.wallendar.models.Charge;

import java.util.HashSet;
import java.util.Set;

public class ChargeMapper {
    public static Charge toModel(ChargeResponse chargeResponse) {
        return new Charge(
                chargeResponse.getTitle(),
                ApplicationUserMapper.toModel(chargeResponse.getOwner()),
                ApplicationUserMapper.toModel(chargeResponse.getDebtors()),
                chargeResponse.getAmount(),
                chargeResponse.getDate()
        );
    }

    public static Set<Charge> toModel(Set<ChargeResponse> chargeResponseList) {
        Set<Charge> charges = new HashSet<>();
        for (ChargeResponse chargeResponse : chargeResponseList) {
            charges.add(toModel(chargeResponse));
        }
        return charges;
    }
}

File: wallendar-app/app/src/main/java/com/td/wallendar/utils/mappers/ChargeMapper.kt
package com.td.wallendar.utils.mappers

import com.td.wallendar.dtos.response.ChargeResponse
import com.td.wallendar.models.Charge
import java.util.*

object ChargeMapper {
    fun toModel(chargeResponse: ChargeResponse): Charge {
        return Charge(
                chargeResponse.id,
                chargeResponse.title,
                ApplicationUserMapper.toModel(chargeResponse.owner),
                ApplicationUserMapper.toModel(chargeResponse.debtors),
                chargeResponse.amount,
                chargeResponse.date
        )
    }

    fun toModel(chargeResponseList: MutableSet<ChargeResponse>): MutableSet<Charge> {
        val charges: MutableSet<Charge> = HashSet()
        for (chargeResponse in chargeResponseList) {
            charges.add(toModel(chargeResponse))
        }
        return charges
    }
}